File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0) return 0;
        sort(nums.begin(),nums.end());
        int n = nums.size();
        int longest = 1;
        int count = 0;
        int lastsmall = INT_MIN;
        for(int i=0;i<n;i++){
            if(nums[i]-1 == lastsmall){
                count++;
                lastsmall= nums[i];
            }
            else if(nums[i]!=lastsmall){
                count=1;
                lastsmall = nums[i];
            }
            longest = max(longest,count);
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        int longest = 1;
        if(n==0) return 0;
        HashSet<Integer> st = new HashSet<>();
        for(int i:nums){
            st.add(i);
        }
        for(int num:st){
            if(!st.contains(num-1)){
                int cnt = 1;
                int x = num;
                while(st.contains(x+1)){
                    x=x+1;
                    cnt++;
                }
                longest = Math.max(longest,cnt);
            }
        }
        return longest;
    }
}